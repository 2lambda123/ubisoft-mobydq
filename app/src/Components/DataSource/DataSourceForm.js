import React from 'react';
import { withFormik } from 'formik';
import * as Yup from 'yup';
import TextInput from './../FormInput/TextInput';
import SelectInput from './../FormInput/SelectInput';
import SaveButton from './../FormInput/SaveButton';
import ExecuteButton from './../FormInput/ExecuteButton';
import DeleteButton from './../FormInput/DeleteButton';

const DataSourceFormFields = (props) => {
  const {
    data,
    values,
    touched,
    errors,
    handleChange,
    handleBlur,
    handleSubmit,
    isSubmitting
  } = props;
  return <form onSubmit={handleSubmit} style={{ 'marginLeft': '50px' }}>
    <div style={{ 'marginTop': '10px', 'marginBottom': '30px' }}>
      <SaveButton disabled={isSubmitting} />
      <ExecuteButton disabled={isSubmitting} />
      <DeleteButton disabled={isSubmitting} />
    </div>
    <div>
      <TextInput
        id="name"
        label="Name"
        helperText=""
        placeholder=""
        touched={touched.name}
        error={touched.name && errors.name}
        value={values.name}
        onChange={handleChange}
        onBlur={handleBlur}
      />
    </div>
    <div>
      <SelectInput
        id="dataSourceTypeId"
        label="Type"
        items={data.allDataSourceTypes.nodes}
        touched={touched.dataSourceTypeId}
        error={touched.dataSourceTypeId && errors.dataSourceTypeId}
        value={values.dataSourceTypeId}
        onChange={handleChange}
        onBlur={handleBlur}
      />
    </div>
    <div>
      <TextInput
        id="connectionString"
        label="Connection string"
        helperText=""
        placeholder="Connection string"
        touched={touched.connectionString}
        error={touched.connectionString && errors.connectionString}
        value={values.connectionString}
        onChange={handleChange}
        onBlur={handleBlur}
        multiline={true}
        rows={5}
      />
    </div>
    <div>
      <TextInput
        id="login"
        label="Login"
        helperText=""
        placeholder=""
        touched={touched.login}
        error={touched.login && errors.login}
        value={values.login}
        onChange={handleChange}
        onBlur={handleBlur}
      />
    </div>
    <div>
      <TextInput
        id="password"
        label="Password"
        helperText=""
        placeholder=""
        touched={touched.password}
        error={touched.password && errors.password}
        value={values.password}
        onChange={handleChange}
        onBlur={handleBlur}
      />
    </div>
    <div>
      <TextInput
        id="connectivityStatus"
        label="Connectivity status"
        helperText=""
        placeholder=""
        touched={touched.connectivityStatus}
        error={touched.connectivityStatus && errors.connectivityStatus}
        value={values.connectivityStatus}
        onChange={handleChange}
        onBlur={handleBlur}
        disabled={true}
        variant={'filled'}
      />
    </div>
    <div>
      <TextInput
        id="createdDate"
        label="Created Date"
        helperText=""
        placeholder=""
        touched={touched.createdDate}
        error={touched.createdDate && errors.createdDate}
        value={values.createdDate}
        onChange={handleChange}
        onBlur={handleBlur}
        disabled={true}
        variant={'filled'}
      />
    </div>
    <div>
      <TextInput
        id="createdBy"
        label="Created By"
        helperText=""
        placeholder=""
        touched={touched.createdBy}
        error={touched.createdBy && errors.createdBy}
        value={values.createdBy}
        onChange={handleChange}
        onBlur={handleBlur}
        disabled={true}
        variant={'filled'}
      />
    </div>
    <div>
      <TextInput
        id="updatedDate"
        label="Updated Date"
        helperText=""
        placeholder=""
        touched={touched.updatedDate}
        error={touched.updatedDate && errors.updatedDate}
        value={values.updatedDate}
        onChange={handleChange}
        onBlur={handleBlur}
        disabled={true}
        variant={'filled'}
      />
    </div>
    <div>
      <TextInput
        id="updatedBy"
        label="Updated By"
        helperText=""
        placeholder=""
        touched={touched.updatedBy}
        error={touched.updatedBy && errors.updatedBy}
        value={values.updatedBy}
        onChange={handleChange}
        onBlur={handleBlur}
        disabled={true}
        variant={'filled'}
      />
    </div>
  </form>
  ;
};

const formikEnhancer = withFormik({
  'validationSchema': Yup.object().shape({
    'name': Yup.string()
      .required('Name cannot be blank'),
    'connectionString': Yup.string()
      .required('Connection string cannot be blank'),
    'login': Yup.string()
      .required('Login cannot be blank')
  }),

  'mapPropsToValues': (props) => props.initialFieldValues === null
    ? {
        'name': '',
        'connectionString': '',
        'dataSourceTypeId': '',
        'login': '',
        'password': ''
      }
    : {
        'name': props.initialFieldValues.name,
        'connectionString': props.initialFieldValues.connectionString,
        'dataSourceTypeId': props.initialFieldValues.dataSourceTypeId,
        'login': props.initialFieldValues.login,
        'password': props.initialFieldValues.password,
        'createdDate': props.initialFieldValues.createdDate,
        'createdBy': props.initialFieldValues.userByCreatedById.email,
        'updatedDate': props.initialFieldValues.updatedDate,
        'updatedBy': props.initialFieldValues.userByUpdatedById.email
      },
  'handleSubmit': (payload, { props, setSubmitting }) => {
    setSubmitting(false);
    let variables;
    if (props.initialFieldValues === null) {
      variables = { 'dataSource': payload };
    } else {
      variables = { 'dataSourcePatch': payload, 'id': props.initialFieldValues.id };
    }
    props.mutate({
      variables
    });
  },
  'displayName': 'DataSourceForm'
});


const EnhancedDataSourceForm = formikEnhancer(DataSourceFormFields);

export default EnhancedDataSourceForm;
