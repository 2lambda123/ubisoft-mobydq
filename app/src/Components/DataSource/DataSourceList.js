import React from 'react';
import { Query } from 'react-apollo';
import DataSourceRepository  from './../../repository/DataSourceRepository';
import ListTable from '../ListTable/ListTable';
import {LinkButton} from './../../Components/FormInput/SimpleButton';

const DataSourceList = (refetch) => (
  <Query
    query={DataSourceRepository.getListPage(1, 10)}
    fetchPolicy={refetch ? 'cache-and-network': 'cache-first'}
  >
    {({ loading, error, data }) => {
      if (loading) return <p>Loading...</p>;
      if (error) return <p>Error :(</p>;
      return (
        <div>
          <div style={{float: 'left', marginLeft: '60px'}}>
            Data Sources
          </div>
          <div style={{float: 'right'}}>
            <LinkButton disabled={false} label="Create" type="Create" color='primary'
              variant='contained' to={'/data-source/new'}/>
          </div>
          <ListTable data={data.allDataSources.nodes}  buttons={[{'function': 'edit', 'parameter': '/data-source'}, {'function': 'delete', 'parameter': DataSourceRepository}]}/>
        </div>
      );
    }}
  </Query>
);

export default DataSourceList;
