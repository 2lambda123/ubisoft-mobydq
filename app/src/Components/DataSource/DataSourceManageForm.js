import React from 'react';

export const DataSourceManageForm = (props) => <div>
  Not yet implemented
</div>