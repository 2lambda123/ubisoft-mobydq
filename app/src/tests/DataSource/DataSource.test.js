import React from 'react';
import { shallowWrap } from './../../setupTests';

import DataSource from './../../Components/DataSource/DataSource';

describe('DataSource component unit test', () => {
  const match = {
    'url': '/data-source'
  };
  const wrapper = shallowWrap(<DataSource match={match}/>);
  it('matches snapshot', () => {
    expect(wrapper.dive()).toMatchSnapshot();
  });
});
