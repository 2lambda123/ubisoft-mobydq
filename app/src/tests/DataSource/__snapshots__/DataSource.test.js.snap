// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataSource component unit test matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DataSource
    classes={
      Object {
        "appBar": "DataSource-appBar-4",
        "appBarShift": "DataSource-appBarShift-5",
        "appBarSpacer": "DataSource-appBarSpacer-11",
        "chartContainer": "DataSource-chartContainer-13",
        "content": "DataSource-content-12",
        "drawerPaper": "DataSource-drawerPaper-9",
        "drawerPaperClose": "DataSource-drawerPaperClose-10",
        "formContainer": "DataSource-formContainer-14",
        "menuButton": "DataSource-menuButton-6",
        "menuButtonHidden": "DataSource-menuButtonHidden-7",
        "root": "DataSource-root-1",
        "tableContainer": "DataSource-tableContainer-15",
        "title": "DataSource-title-8",
        "toolbar": "DataSource-toolbar-2",
        "toolbarIcon": "DataSource-toolbarIcon-3",
      }
    }
    match={
      Object {
        "url": "/data-source",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <div
          className="DataSource-appBarSpacer-11"
        />,
        <WithStyles(Typography)
          className="DataSource-chartContainer-13"
          gutterBottom={true}
          variant="display1"
        >
          <Route
            component={[Function]}
            path="/data-source/new"
          />
          <Route
            component={[Function]}
            path="/data-source/edit/:id"
          />
          <Route
            exact={true}
            path="/data-source"
            render={[Function]}
          />
        </WithStyles(Typography)>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "DataSource-appBarSpacer-11",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Route
              component={[Function]}
              path="/data-source/new"
            />,
            <Route
              component={[Function]}
              path="/data-source/edit/:id"
            />,
            <Route
              exact={true}
              path="/data-source"
              render={[Function]}
            />,
          ],
          "className": "DataSource-chartContainer-13",
          "gutterBottom": true,
          "variant": "display1",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/data-source/new",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/data-source/edit/:id",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "exact": true,
              "path": "/data-source",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <div
            className="DataSource-appBarSpacer-11"
          />,
          <WithStyles(Typography)
            className="DataSource-chartContainer-13"
            gutterBottom={true}
            variant="display1"
          >
            <Route
              component={[Function]}
              path="/data-source/new"
            />
            <Route
              component={[Function]}
              path="/data-source/edit/:id"
            />
            <Route
              exact={true}
              path="/data-source"
              render={[Function]}
            />
          </WithStyles(Typography)>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "DataSource-appBarSpacer-11",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Route
                component={[Function]}
                path="/data-source/new"
              />,
              <Route
                component={[Function]}
                path="/data-source/edit/:id"
              />,
              <Route
                exact={true}
                path="/data-source"
                render={[Function]}
              />,
            ],
            "className": "DataSource-chartContainer-13",
            "gutterBottom": true,
            "variant": "display1",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/data-source/new",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/data-source/edit/:id",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "exact": true,
                "path": "/data-source",
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "childContextTypes": Object {
      "router": [Function],
    },
    "context": Object {
      "router": Object {
        "history": Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        },
        "route": Object {
          "location": Object {},
          "match": Object {},
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
